ios/Capacitor/Capacitor/Plugins/CapacitorUrlRequest.swift

```diff
@@ -93,7 +93,65 @@ open class CapacitorUrlRequest: NSObject, URLSessionTaskDelegate {
         return normalized[index.lowercased()]
     }
 
-    public func getRequestData(_ body: JSValue, _ contentType: String) throws -> Data? {
+    func getRequestDataFromFormData(_ data: JSValue) throws -> Data? {
+        guard let list = data as? JSArray else {
+            // Throw, other data types explicitly not supported.
+            throw CapacitorUrlRequestError.serializationError("Data must be an array for FormData")
+        }
+
+        var data = Data()
+
+        // Update the contentType with the new boundary
+        let boundary = UUID().uuidString
+        let contentType = "multipart/form-data; boundary=\(boundary)"
+        request.setValue(contentType, forHTTPHeaderField: "Content-Type")
+        headers["Content-Type"] = contentType
+
+        for entry in list {
+            guard let item = entry as? [String: String] else {
+                throw CapacitorUrlRequestError.serializationError("Data must be an array for FormData")
+            }
+
+            let type = item["type"]
+            let key = item["key"]
+            let value = item["value"]!
+
+            if type == "base64File" {
+                let fileName = item["fileName"]
+                let fileContentType = item["contentType"]
+
+                data.append("\r\n--\(boundary)\r\n".data(using: .utf8)!)
```

Comment:
There has to be a better way to do this, but I can't think of one of the top of my head - maybe just creating one string and converting that? But that'll be more confusing. 

That said, check out this comment that may require adding some lines:
https://github.com/ionic-team/capacitor/pull/6206#issuecomment-1594861070

Reply:
It would be nice if there was a better abstraction around form data, like there is on the javascript side, but this does get the job done...